src/charm.py

```diff
@@ -407,11 +411,30 @@ def _gen_certs(self):
 
         return ret_certs
 
+    def _process_gateway_info(self):
+        """Parse gateway information and store it, if it is not empty."""
+        try:
+            gateway_info = self._gateway.get_relation_data()
+        except GatewayRelationError:
+            self.model.unit.status = WaitingStatus("Waiting for gateway info relation")
+            return
+        if gateway_info["gateway_namespace"] and gateway_info["gateway_name"]:
+            self._gateway_info["namespace"] = gateway_info["gateway_namespace"]
+            self._gateway_info["name"] = gateway_info["gateway_name"]
```

Comment:
This comment makes more sense after reading the previous one about `_get_istio_gateway`.  This method felt more like a `get` than `process`.  It fetches the data and stores it, so I think it is closer to `get_gateway_info` or `fetch_gateway_info`.  What do you think?

Reply:
Could be. For me `get` implies returning something and here we are basically storing data in different format.
Do we really fetch/get anything here? How is `GatewayRequirer:_gateway` gets populated?
In this case data is already stored (by who?) in `self._gateway` and we are actually processing it: taking it out of stored format and converting it into our own format.
If that `sefl._gateway` is empty, then we indicate that we are still waiting for it to be populated.